channel/backend.go

```diff
@@ -152,7 +158,10 @@ func Verify(addr wallet.Address, s *channel.State, sig wallet.Sig) (bool, error)
 func ToEthParams(p *channel.Params) adjudicator.ChannelParams {
 	var app common.Address
 	if p.App != nil && !channel.IsNoApp(p.App) {
-		app = ethwallet.AsEthAddr(p.App.Def())
+		appDef, ok := p.App.Def().(*AppID)
```

Comment:
Just asking: Is it intentional that types that don't match are silently dropped, no panic, no error, no warning? Could lead to quite confusing debugging sessions if app is suddenly the 0x00 address because the type didn't match exactly here (I assume the type is supposed to always match).